pandas/core/internals/blocks.py

```diff
@@ -352,10 +352,6 @@ def fillna(self, value, limit=None, inplace=False, downcast=None):
 
         mask = isna(self.values)
         if limit is not None:
-            if not is_integer(limit):
```

Comment:
This is removing validation in this method right? Shouldn't we replace with a call to `validate_fillna_kwargs`?

Reply:
Digging into the fillna callstack, it first calls super().fillna() in core/frame.py, which then goes to core/generic.py where validate_fillna_kwargs is called (along with validate_bool_kwarg on `inplace`), before it then goes into core/internalls/blocks.py. Then in fillna in blocks.py it calls validate_bool_kwarg on `inplace` again, but originally never did the validate_fillna_kwargs call again. I'm not sure if the checks are just being repeated needlessly or not, first in frame and then in blocks. But at the start of the function validate_fillna_kwargs(value, None, limit) could be called I guess.